src/compiler/program.ts

```diff
@@ -436,6 +436,7 @@ namespace ts {
         host = host || createCompilerHost(options);
 
         let skipDefaultLib = options.noLib;
+        const defaultLibraryPath = host.getDefaultLibLocation ? host.getDefaultLibLocation() : getDirectoryPath(host.getDefaultLibFileName(options));

```

Comment:
There is also a `libDirectory` variable in this function that does the same thing. Consider combining the two and possibly storing the result of `host.getDefaultLibFileName(options)` in a variable rather than requery it each time.